controller/users.go

```diff
@@ -49,6 +50,102 @@ func NewUsersController(service *goa.Service, db application.DB, config UsersCon
 		config:     config,
 	}
 }
+func randString(n int) string {
+	const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Int63()%int64(len(letterBytes))]
+	}
+	return string(b)
+}
+
+// Obfuscate runs the obfuscate action to soft delete sensitive data associated
+// with an user and her associated identity.
+func (c *UsersController) Obfuscate(ctx *app.ObfuscateUsersContext) error {
+	isSvcAccount, err := isServiceAccount(ctx, serviceNameAuth)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to determine if account is a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err))
+
+	}
+	if !isSvcAccount {
+		log.Error(ctx, map[string]interface{}{
+			"identity_id": ctx.ID,
+		}, "account used to call create api is not a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(errs.New("a non-service account tried to create a user.")))
+	}
+	u, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to convert user id in valid uuid")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest(errs.New("invalid user id")))
+
+	}
+	returnResponse := application.Transactional(c.db, func(appl application.Application) error {
```

Comment:
@corinnekrych We'll have to change this. The usual way to use `application.Transactional` block is
```go
err := application.Transactional(c.db, func(appl application.Application) error {
	// do some database stuff
	user, err := appl.Users().Load(ctx, UserID)
	if err != nil {
		return errs.Wrap(err, "failed to load user")
	}
})
// database stuff failed
if err != nil {
	// return error response
}
// return success response
```

The problem with your approach is you're returning errors (like https://github.com/fabric8-services/fabric8-wit/pull/2358/files#diff-da062b312519ee2a61dd669f1ec964d4R112) directly to user (the `returnResponse` will be a database error). We should avoid that :)